src/core/libraries/network/net.cpp

```diff
@@ -644,17 +646,33 @@ int PS4_SYSV_ABI sceNetEpollControl(OrbisNetId epollid, OrbisNetEpollFlag op, Or
             return ORBIS_NET_ERROR_EEXIST;
         }
 
-        auto file = FDTable::Instance()->GetSocket(id);
+        auto file = FDTable::Instance()->GetFile(id);
         if (!file) {
             *sceNetErrnoLoc() = ORBIS_NET_EBADF;
-            LOG_ERROR(Lib_Net, "socket id is invalid = {}", id);
+            LOG_ERROR(Lib_Net, "file id is invalid = {}", id);
             return ORBIS_NET_ERROR_EBADF;
         }
-        epoll_event native_event = {.events = ConvertEpollEventsIn(event->events),
-                                    .data = {.fd = id}};
-        ASSERT(epoll_ctl(epoll->epoll_fd, EPOLL_CTL_ADD, *file->socket->Native(), &native_event) ==
-               0);
-        epoll->events.emplace_back(id, *event);
+
+        switch (file->type) {
+        case Core::FileSys::FileType::Socket: {
+            epoll_event native_event = {.events = ConvertEpollEventsIn(event->events),
+                                        .data = {.fd = id}};
+            ASSERT(epoll_ctl(epoll->epoll_fd, EPOLL_CTL_ADD, *file->socket->Native(),
```

Comment:
Testing with The Little big planet 3, I'm having a stop right here because the socket is P2P. What I've seen nothing is implemented right now for that so the value of std::optional.has_value() is false.

I think things should be :

```suggestion
            if(file->socket->Native().has_value()){
                int res = epoll_ctl(epoll->epoll_fd, EPOLL_CTL_ADD, *file->socket->Native(), &native_event);
                ASSERT_MSG(res == 0,
                       "epoll_ctl(ADD) failed for socket fd {}, error code {}", errno);
            } else {
                ASSERT_MSG(false, "Socket has no native handle {}", file->m_guest_name);
            }
```

Reply:
Thanks @aaalloc, you're correct P2P sockets are not implemented right now, that is the next thing I'm working on